src/components/Contentful/ContentfulDynamicHero.vue

```diff
@@ -0,0 +1,229 @@
+			>
+				<template v-if="isHeroImage">
+					<router-link
+						:to="heroButton.link"
+						v-kv-track-event="[
+							'homepage',
+							'click-hero-loancards',
+							heroMedia.description,
+						]"
+					>
+						<kv-contentful-img
+							v-if="heroMedia.url"
+							class="contentful-hero__img"
+							:contentful-src="heroMedia.url"
+							:width="500"
+							:height="527"
+							fallback-format="jpg"
+							:alt="heroMedia.description"
+						/>
+					</router-link>
+				</template>
+				<template v-if="isHeroVideo">
+					<video
+						:src="heroMedia.url"
+						class="contentful-hero__video-wrapper--video"
+						autoplay
+						loop
+						muted
+						playsinline
+						height="400"
+					></video>
+				</template>
+			</div>
+			<!-- eslint-disable-next-line max-len -->
+			<div class="small-10 large-6 xlarge-7 align-self-middle columns contentful-hero__cta_wrapper">
+				<h1 class="contentful-hero__header" v-html="heroHeadline">
+				</h1>
+				<div class="contentful-hero__body" v-html="heroBody">
+				</div>
+				<kv-button
+					:class="`${buttonClass} show-for-large rounded`"
+					:to="buttonTo"
+					@click.native="buttonClick"
+					v-kv-track-event="[
+						'homepage',
+						'click-hero-cta',
+						heroButton.text,
+					]"
+				>
+					{{ heroButton.text }}
+				</kv-button>
+			</div>
+		</div>
+		<!-- Button in different element order for mobile only -->
+		<div class="row align-center hide-for-large">
+			<div class="small-10">
+				<kv-button
+					:class="`${buttonClass} rounded expanded`"
+					:to="buttonTo"
+					@click.native="buttonClick"
+					v-kv-track-event="[
+						'homepage',
+						'click-hero-cta',
+						heroButton.text,
+					]"
+				>
+					{{ heroButton.text }}
+				</kv-button>
+			</div>
+		</div>
+	</section>
+</template>
+
+<script>
+import KvButton from '@/components/Kv/KvButton';
+import KvContentfulImg from '@/components/Kv/KvContentfulImg';
+import { documentToHtmlString } from '~/@contentful/rich-text-html-renderer';
+
+export default {
+	components: {
+		KvButton,
+		KvContentfulImg,
+	},
+	props: {
+		content: {
+			type: Object,
+			default: () => {},
+		},
+	},
+	computed: {
+		mediaObject() {
+			return this.content?.media?.[0];
+		},
+		isHeroVideo() {
+			return this.mediaObject?.file?.contentType.includes('video');
+		},
+		isHeroImage() {
+			return this.mediaObject?.file?.contentType.includes('image');
+		},
+		buttonClass() {
+			return this.$attrs?.customCtaButtonClass ?? '';
+		},
+		buttonTo() {
+			if (this.$attrs?.customCtaFunction) {
+				return null;
+			}
+			return this.heroButton.link;
+		},
+		buttonClick() {
+			return this.$attrs?.customCtaFunction ?? null;
+		},
+		heroMedia() {
+			return {
+				description: this.mediaObject?.description ?? '',
+				title: this.mediaObject?.title ?? '',
+				url: this.mediaObject?.file?.url ?? ''
+			};
+		},
+		heroText() {
+			return this.content?.contents?.find(({ key }) => key.indexOf('homepage-hero-text') > -1);
```

Comment:
Should we change this to be more generic?

Reply:
Seems like we'd want this to remain as homepage specific....

Reply:
This is a Generic Content Block, not a Content Group, so we don't have a type field that we can key off of.

I think it would be fine to change this to be checking the key for just 'hero-text'.